File: algorithms/3_hard/M/02366_minimum_replacements_to_sort_the_array/greedy.c
long long minimumReplacement(const int *restrict nums, const int len) {
    int min = INT_MAX;
    int64_t cnt = 0;

    for (int i = len - 1; i >= 0; i -= 1) {
        const int num = nums[i];

        if (num <= min) {
            min = num;
        } else {
            const int parts = (num + min - 1) / min;
            min = num / parts;
            cnt += parts - 1;
        }
    }

    return cnt;
}

File: algorithms/3_hard/M/02366_minimum_replacements_to_sort_the_array/greedy.rs
impl Solution {
    pub fn minimum_replacement(nums: Vec<i32>) -> i64 {
        let mut min = i32::MAX;
        let mut cnt: i64 = 0;

        for num in nums.into_iter().rev() {
            if num <= min {
                min = num;
            } else {
                let parts = (num + min - 1) / min;
                min = num / parts;
                cnt += (parts - 1) as i64;
            }
        }

        cnt
    }
}
